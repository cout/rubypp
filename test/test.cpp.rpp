#include <iostream>

#ruby <<END
  puts '#define FOO BAR'
  '#define BAR BAZ'
END

#ruby def foo(x)              ; \
        x.to_s.chop           ; \
      end

extern "C" {
int foo(int a) {
  std::cout << "1" << std::endl;
}

int foo(double a) {
  std::cout << "2" << std::endl;
}

int main() {
  foo(1);
  foo(1.0);
  std::cout << "#{foo(1.0)}" << std::endl;
}

