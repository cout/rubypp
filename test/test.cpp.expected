#include <iostream>

#define FOO BAR
#define BAR BAZ


extern "C" {
int foo(int a) {
  std::cout << "1" << std::endl;
}

int foo(double a) {
  std::cout << "2" << std::endl;
}

int main() {
  foo(1);
  foo(1.0);
  std::cout << "1." << std::endl;
}

